src/kOS.Safe/Encapsulation/Structure.cs

```diff
@@ -337,20 +337,10 @@ public static Structure FromPrimitiveWithAssert(object value)
 
         public static object ToPrimitive(object value)
         {
-            var scalarValue = value as ScalarValue;
-            if (scalarValue != null)
+            var primitive = value as PrimitiveStructure;
+            if (primitive != null)
             {
-                return scalarValue.Value;
-            }
-            var booleanValue = value as BooleanValue;
-            if (booleanValue != null)
-            {
-                return booleanValue.Value;
-            }
-            var stringValue = value as StringValue;
-            if (stringValue != null)
-            {
-                return stringValue.ToString();
+                return primitive.ToPrimitive();
```

Comment:
The reason I originally left the `ToPrimitive` method as a static method on the `Structure` class was because we cannot implement `FromPrimitive` on the individual structures.  While I agree that this logic is better suited to be in the individual classes (since it's part of the class definition) I rather liked the fact that both `From` and `To` were in the same place, making it so that edits to the system only needed to be done in a single file.  Was this done to facilitate easier serialization?  Does it maybe make sense to treat the on structure `ToPrimitive` logic strictly in a serialization context?